pipelines/plain_pipelines/paas-hackmd.yml

```diff
@@ -0,0 +1,66 @@
+---
+resource_types:
+  - name: s3-iam
+    type: docker-image
+    source:
+      repository: governmentpaas/s3-resource
+      tag: fda60bf4c5f85e96c16f704e128e5ead9e84d30d
+
+resources:
+  - name: paas-codimd
+    type: git
+    source:
+      uri: https://github.com/alphagov/paas-codimd
+      #FIXME: pin to gds_master
+      branch: initial
+
+jobs:
+  - name: deploy
+    serial: true
+    plan:
+      - get: paas-codimd
+        trigger: true
+      - task: push
+        config:
+          platform: linux
+          image_resource:
+            type: docker-image
+            source:
+              repository: governmentpaas/cf-cli
+              tag: 0eff5b6a9c092f865a2b19cc4e75a3b539b82fa2
+          inputs:
+            - name: paas-codimd
+          params:
+            CF_API: https://api.((cf_system_domain))
+            CF_USER: ((cf_user))
+            CF_PASSWORD: ((cf_password))
+            CF_APPS_DOMAIN: ((cf_apps_domain))
+            CF_SYSTEM_DOMAIN: ((cf_system_domain))
+            CF_ORG: govuk-paas
+            CF_SPACE: tools
+          run:
+            path: sh
+            args:
+              - -e
+              - -u
+              - -c
+              - |
+                echo "Logging on to Cloudfoundry..."
+                cf login \
+                  -a "${CF_API}" \
+                  -u "${CF_USER}" \
+                  -p "${CF_PASSWORD}" \
+                  -o "${CF_ORG}" \
+                  -s "${CF_SPACE}"
+
+                cd paas-codimd/
+                cf push --no-start
+
+                cf create-service postgres tiny-unencrypted-9.5 hackmd-db
+                while ! cf service hackmd-db | grep -q 'create succeeded'; do
+                  echo "Waiting for creation of service to complete..."
+                  sleep 30
+                done
+
+                cf bind-service hackmd hackmd-db
+                cf restart hackmd
```

Comment:
It would be good to make this use a blue-green deployment to avoid downtime... See `paas-tech-docs.yml` for an example - the necessary plugin is already in the docker image.

Reply:
Would that account for the required database binding?